index.ts

```diff
@@ -65,134 +68,133 @@ function newContext() {
           if (nextObject === object) {
             nextObject = copy(object);
           }
           if (type(nextObject) === 'Map') {
-            nextObject.set(key, nextValueForKey);
+            (nextObject as any as Map<any, any>).set(key, nextValueForKey);
           } else {
             nextObject[key] = nextValueForKey;
           }
         }
       }
-    })
+    });
     return nextObject;
   }
-
 }
 
-var defaultCommands = {
-  $push: function(value, nextObject, spec) {
+const defaultCommands = {
+  $push: (value, nextObject, spec) => {
     invariantPushAndUnshift(nextObject, spec, '$push');
     return value.length ? nextObject.concat(value) : nextObject;
   },
-  $unshift: function(value, nextObject, spec) {
+  $unshift: (value, nextObject, spec) => {
     invariantPushAndUnshift(nextObject, spec, '$unshift');
     return value.length ? value.concat(nextObject) : nextObject;
   },
-  $splice: function(value, nextObject, spec, originalObject) {
+  $splice: (value, nextObject, spec, originalObject) => {
     invariantSplices(nextObject, spec);
-    value.forEach(function(args) {
+    value.forEach(args => {
       invariantSplice(args);
       if (nextObject === originalObject && args.length) nextObject = copy(originalObject);
       splice.apply(nextObject, args);
     });
     return nextObject;
   },
-  $set: function(value, nextObject, spec) {
+  $set: (value, _nextObject, spec) => {
     invariantSet(spec);
     return value;
   },
-  $toggle: function(targets, nextObject) {
+  $toggle: (targets, nextObject) => {
     invariantSpecArray(targets, '$toggle');
-    var nextObjectCopy = targets.length ? copy(nextObject) : nextObject;
+    const nextObjectCopy = targets.length ? copy(nextObject) : nextObject;
 
-    targets.forEach(function(target) {
+    targets.forEach(target => {
       nextObjectCopy[target] = !nextObject[target];
     });
 
     return nextObjectCopy;
   },
-  $unset: function(value, nextObject, spec, originalObject) {
+  $unset: (value, nextObject, _spec, originalObject) => {
     invariantSpecArray(value, '$unset');
-    value.forEach(function(key) {
+    value.forEach(key => {
       if (Object.hasOwnProperty.call(nextObject, key)) {
         if (nextObject === originalObject) nextObject = copy(originalObject);
+        // tslint:disable-next-line:no-dynamic-delete
         delete nextObject[key];
       }
     });
     return nextObject;
   },
-  $add: function(value, nextObject, spec, originalObject) {
+  $add: (value, nextObject, _spec, originalObject) => {
     invariantMapOrSet(nextObject, '$add');
     invariantSpecArray(value, '$add');
     if (type(nextObject) === 'Map') {
-      value.forEach(function(pair) {
-        var key = pair[0];
-        var value = pair[1];
+      value.forEach(pair => {
+        const key = pair[0];
+        const value = pair[1];
+        /* istanbul ignore next */
```

Comment:
This is the only ignore line I added and I have no idea why I had to. @kolodny do you know which test this relates to? I didn't add or remove any tests.

Reply:
I'm not sure, this is ignoring this line, so we should have better test coverage to try to catch this line. A lot of folks are into forcing 100% coverage especially for library code but I'm OK with it a little lower. My preference would be to remove since this is something we could cover unlike the polyfilled `Object.assign` or some other polyfill code. Up to you either way